src/Components/GEOSldas_GridComp/GEOSlandassim_GridComp/clsm_ensupd_read_obs.F90

```diff
@@ -7645,6 +8156,283 @@ subroutine scale_obs_tskin_zscore( N_catd, tile_coord,   &
+    ierr = nf90_inquire_dimension(ncid, lon_dimid,    len = N_lon)
+    ierr = nf90_inquire_dimension(ncid, lat_dimid,    len = N_lat)
+    
+    ! Get the variable IDs
+
+    ierr = nf90_inq_varid(ncid, 'lon',    lon_varid)
+    ierr = nf90_inq_varid(ncid, 'lat',    lat_varid)
+    ierr = nf90_inq_varid(ncid, 'o_mean', o_mean_varid)
+    ierr = nf90_inq_varid(ncid, 'o_std',  o_std_varid)
+    ierr = nf90_inq_varid(ncid, 'm_mean', m_mean_varid)
+    ierr = nf90_inq_varid(ncid, 'm_std',  m_std_varid)
+    ierr = nf90_inq_varid(ncid, 'm_min',  m_min_varid)
+    ierr = nf90_inq_varid(ncid, 'm_max',  m_max_varid)    
+    
+    ! Read grid variables
+
+    ierr = nf90_get_var(ncid, ll_lon_varid, ll_lon)
+    ierr = nf90_get_var(ncid, ll_lat_varid, ll_lat)
+    ierr = nf90_get_var(ncid, dlon_varid,   dlon)
+    ierr = nf90_get_var(ncid, dlat_varid,   dlat)
+    
+    ! Read lon and lat variables
+
+    allocate(sclprm_lon(N_lon), sclprm_lat(N_lat))
+
+    ierr = nf90_get_var(ncid, lon_varid, sclprm_lon)
+    ierr = nf90_get_var(ncid, lat_varid, sclprm_lat)
+    
+    start  = [1,     1,     pp]
+    icount = [N_lat, N_lon, 1 ]
+    
+    ! Read mean and std variables
+
+    allocate(sclprm_mean_obs(N_lat, N_lon), sclprm_std_obs(N_lat, N_lon))
+    allocate(sclprm_mean_mod(N_lat, N_lon), sclprm_std_mod(N_lat, N_lon))
+    allocate(sclprm_min_mod( N_lat, N_lon), sclprm_max_mod(N_lat, N_lon))
+
+    ierr  = nf90_get_var(ncid, o_mean_varid, sclprm_mean_obs, start, icount)
+    ierr  = nf90_get_var(ncid, o_std_varid,  sclprm_std_obs,  start, icount)
+    ierr  = nf90_get_var(ncid, m_mean_varid, sclprm_mean_mod, start, icount)
+    ierr  = nf90_get_var(ncid, m_std_varid,  sclprm_std_mod,  start, icount)
+    ierr  = nf90_get_var(ncid, m_min_varid,  sclprm_min_mod)
+    ierr  = nf90_get_var(ncid, m_max_varid,  sclprm_max_mod)
+
+    ! Close the netcdf file
+
+    ierr = nf90_close(ncid)
+    
+    ! --------------------------------------------------------------
+    
+    ! Scale observations (at this point all obs are of same type because
+    ! of the way the subroutine is called from subroutine read_obs()
+    
+    do i=1,N_catd
+       
+       ! Check for no-data-values in observation (any neg value is no_data)
+
+       if (tmp_obs(i)>=0.) then
+          
+          ! ll_lon and ll_lat refer to lower left corner of grid cell
+          ! (as opposed to the grid point in the center of the grid cell)
+          
+          this_lon = tmp_lon(i)
+          this_lat = tmp_lat(i)
+          
+          ! Find indices for current tile lat and lon on scaling parameter grid
+
+          i_ind = ceiling((this_lon - ll_lon)/dlon) 
+          j_ind = ceiling((this_lat - ll_lat)/dlat) 
+          
+          ! Sanity check (against accidental use of wrong tile space)
+
+          if ( abs(tile_coord(i)%com_lat-sclprm_lat(j_ind))>tol  .or.             &
+               abs(tile_coord(i)%com_lon-sclprm_lon(i_ind))>tol ) then
```

Comment:
This check compares the tile's center-of-mass with the  **lower boundary** of the scaling parameter grid cell.  It would make more sense to have 'lon' and 'lat' indicate the lon & lat of the **center** of the scaling parameter grid cells.  I think this is the only place that uses 'lon' and 'lat' from the netcdf scaling param file.  We may not even need this check.

Reply:
So having looked at where tmp_lon and tmp_lat come from it doesn't seem like we really need to keep this check. 

There is definitely some circularity as although tmp_lon and tmp_lat are derived solely from observation values, the observations themselves are initially assigned to a tile based on `tile_coord%com_lat` or `tile_coord%com_lon` in `get_tile_num_from_latlon`, so I don't really see how this test could ever fail. I guess its definitely checking nothing has gone wrong in the "super ob-ing" stage.

But if we want to keep it we wouldn't need to use the lat and lon vectors read from the netCDF file, we could just calculate it at the time given the grid info.
Something like: 
         
```
          this_lon = tmp_lon(i)
          this_lat = tmp_lat(i)
          
          ! Find indices for current tile lat and lon on scaling parameter grid

          i_ind = ceiling((this_lon - ll_lon)/dlon) 
          j_ind = ceiling((this_lat - ll_lat)/dlat) 

          sclprm_lon = (ll_lon - dlon/2) + (i_ind*dlon)
          sclprm_lat = (ll_lat - dlat/2) + (j_ind*dlat)
                    
          ! Sanity check (against accidental use of wrong tile space)

          if ( abs(tile_coord(i)%com_lat-sclprm_lat)>tol  .or.             &
               abs(tile_coord(i)%com_lon-sclprm_lon)>tol ) then
                 
               err_msg = 'Lat/lon diff beyond tolerance'
               call ldas_abort(LDAS_GENERIC_ERROR, Iam, err_msg)

          end if
```

Reply:
Thanks, @amfox37, for taking another look at this.  I agree with the assessment.  When you get a chance, please remove the sanity check and remove lat/lon from the scaling parameter file.

Reply:
@gmao-rreichle  Ok, have done this in commit [544a119](https://github.com/GEOS-ESM/GEOSldas/pull/656/commits/544a119cdf58f83b37204028c97eeea77d9ce9c5) removing lat/lon matlab write, lat/lon FORTRAN read and sanity check in read_obs.
Basic testing suggest this change, and all other changes in matlab scripts are good to go.